src/tools/blast/blastcommandline.jl

```diff
@@ -26,32 +26,29 @@ readblastXML(results)
 
 Returns Vector{BLASTResult} with the sequence of the hit, the Alignment with query sequence, bitscore and expect value
 """
-function readblastXML(blastrun::AbstractString; seqtype="nucl")
-    xdoc = parse_string(blastrun)
-    xroot = root(xdoc)
-    params = get_elements_by_tagname(xroot, "BlastOutput_param")
-    iterations = get_elements_by_tagname(xroot, "BlastOutput_iterations")
-    results = BLASTResult[]
-    for iteration in collect(child_elements(iterations[1]))
-        queryname = content(find_element(iteration, "Iteration_query-def"))
-        hits = get_elements_by_tagname(iteration, "Iteration_hits")
-        for hit in collect(child_elements(hits[1]))
-            hitname = content(find_element(hit, "Hit_def"))
-            hsps = get_elements_by_tagname(hit, "Hit_hsps")
-            for hsp in collect(child_elements(hsps[1]))
+function readblastXML(blastrun::String; seqtype="nucl")
+    dc = parsexml(blastrun)
+    rt = root(dc)
+    results = Vector{BLASTResult}([])
+    for iteration in find(rt, "//BlastOutput_iterations/Iteration")
```

Comment:
`find(rt, "/BlastOutput/BlastOutput_iterations/Iteration")` is more explicit and reduces the search space.